### ISSUE TITLE
Invalid JSON after update during email send.

### ISSUE THREAD

**Issue by cbtilf** [formtools/core#581]
Since textarea fields do not validate the same as textbox fields in this Smarty implementation, I had to work around my output to send via email.  This worked in Formtools v.3.0.13 and below.  I have to use a textarea to have a larger block of text to show when receiving input into the field.  But now I am getting this error when trying to generate the email that worked before v.3.0.14.

> 
> Error: Invalid JSON: 
> Fatal error: Uncaught --> Smarty Compiler: Syntax error in template "eval:</br> Created: {$SUBMISSIONDATE} </br> ..." on line 93 "{if ($ANSWER_Issue2|count_characters) > 5}" unknown modifier "count_characters" <-- thrown in /mnt/array1/Internal/htdocs/formtools/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatecompilerbase.php on line 93

This is the original code that used to work with no issue.

```
{if ($ANSWER_Issue2|count_characters) > 0}
  <tr>
    <td style="font-weight: bold" valign="top">Issue 2:				</td>
    <td valign="top">{$ANSWER_Issue2}</td>
  </tr>
{/if}
```
This would output the "Issue 2: + {$ANSER_Issue2)" text if anything was there or skip it if not.

I was able to downgrade to 3.0.13 although it says it's still 3.0.15.  So I'm not sure how stable everything is, so I'd like to get this fixed so I can get future updates.

I had to use the character count function because {if $ANSWER_Issue2} would not validate if the if that field was a textarea instead of a textbox.  Please help me get this working again.  I can not find another workaround to validate if anything is in the textbox.  The standard isset php functions will not work on a textarea, at least not that I've been able to.   I am using the Swift Mailer module.  Thanks!

**Comment by benkeen** [comment]
Hmm, that's puzzling that's changed in recent versions... count_characters is a baked-in Smarty modifier so it should be available anywhere. Let me check it out & get back to you - it may be a couple of days, I'm afraid. Rather struggling to find time for this project at the moment.

**Comment by cbtilf** [comment]
I appreciate it.  It's working on my critical system for now as long as I don't upgrade.  I am testing on a second system right now, so I was able to duplicate the issue and haven't been able to resolve yet.  Please just look into it when you have a chance.  Thanks!

**Comment by cbtilf** [comment]
I just wanted to clarify that it was only Core update I did from 3.0.13 where the problem started.  I first went to 3.0.14 and figured it was a bug then went to 3.0.15.  On my second machine I made sure to update all the modules at each step, and none of those made a difference.  It seemed strange to me as well.  Thanks!

**Comment by benkeen** [comment]
Hey @cbtilf, yeah this a bug, apologies. I've flagged this to be fixed in the next release but I have a mountain of work to do before it'll be out the door. So if you fancy doing a little brain surgery, here's how you can fix it in the meantime... 

Edit the `[form tools root]/global/code/Emails.class.php`. On line 458, you'll see this:

```php
$smarty->setPluginsDir(array(
    "$root_dir/global/smarty_plugins"
));
```

Change `setPluginsDir` to `addPluginsDir`. That's it. It was incorrectly overwriting the default plugins directory that contains the out-the-box plugins. 

> Note to self: when patching this issue, review all other Smarty instance creation in the application.

**Comment by benkeen** [comment]
Let me know if this fixes it for you & I'll close this ticket.

**Comment by cbtilf** [comment]
I found that on 478 when I opened it, but that may just be my editor.  I was opening the file for 3.0.15.  And,  yes, it worked!!  Since my critical version is in a weird state.  It is really 3.0.13 core but says it's 3.0.15 and the modules aren't showing descriptions and say they need updating but won't, what is the best way for me to get it fully up to 3.0.15?  Do I just need to start from the backup I made of the formtools directory before I tried to update to 3.0.14?
